### ISSUE TITLE
Race condition between stream creation and MSPC

### ISSUE THREAD

**Issue by ekr** [quicwg/base-drafts#200]
Consider the following sequence of events on a stream (assuming that MSPC allows exactly one application data stream in the C-S direction:

```
<---------- Stream Establishment ----------->
<----------------------------------------- FIN
FIN ----------------------------------------->
DATA (stream N+1) -------------------------->
```

If you get reordering between the FIN and the data then the server will see it as exceeding MSPC and send  QUIC_TOO_MANY_OPEN_STREAMS. Ugh.

**Comment by Aron-Schats** [comment]
The client should not do that:

https://groups.google.com/a/chromium.org/forum/#!topic/proto-quic/27ee0fWisTc

**Comment by ekr** [comment]
Well, that needs to go in the spec then.

**Comment by martinthomson** [comment]
This was addressed in #421.  Reopen if you disagree.
